app/Utils/CheckoutHandler.php

```diff
@@ -209,6 +212,9 @@ public function addExpense(Request $request): RedirectResponse
             'paid_at'           => $paid ? Carbon::now() : null,
         ]);
 
+        $path = $request->file('receipt')->store('receipts');
```

Comment:
Wrap this and the Transaction::create in a DB::transaction(...), so if either of the logic fails it backtracks. Same with deletion.